pkg/common/peertracker/tcp_windows.go

```diff
@@ -0,0 +1,51 @@
+//go:build windows
+// +build windows
+
+package peertracker
+
+/*
+#cgo LDFLAGS: -liphlpapi -lws2_32
+#include <getpid.h>
+*/
+import "C"
+
+import (
+	"fmt"
+	"net"
+	"syscall"
+
+	"golang.org/x/sys/windows"
+)
+
+func getCallerInfoFromTCPConn(conn net.Conn) (CallerInfo, error) {
+	agentAddr, ok := conn.LocalAddr().(*net.TCPAddr)
+	if !ok {
+		return CallerInfo{}, ErrInvalidConnection
+	}
+
+	callerAddr, ok := conn.RemoteAddr().(*net.TCPAddr)
+	if !ok {
+		return CallerInfo{}, ErrInvalidConnection
+	}
+
+	pid := C.getOwningPIDFromLocalConn(C.int(callerAddr.Port), C.int(agentAddr.Port))
+	if pid < 0 {
+		return CallerInfo{}, fmt.Errorf("failed to get owning PID. Return code is: %d", pid)
+	}
+
+	return CallerInfo{
+		Addr: conn.RemoteAddr(),
+		PID:  int32(pid),
+	}, nil
+}
+
+func compareObjectHandles(firstHandle, secondHandle windows.Handle) (err error) {
+	modkernel32 := windows.NewLazySystemDLL("kernelbase.dll")
+	procCompareObjectHandles := modkernel32.NewProc("CompareObjectHandles")
```

Comment:
Can we move these to function level variables? Reloading the DLL on every comparison seems unnecessary.